secure/storage/vault/src/lib.rs

```diff
@@ -502,6 +500,20 @@ pub fn process_transit_export_response(
     }
 }
 
+/// Fetches the Ed25519 private key from a composite key (i.e., the structure returned by
+/// a vault transit export key request). Composite key = [private|public]
+fn get_ed25519_private_key_from_composite(composite_key: &str) -> Result<Ed25519PrivateKey, Error> {
+    let composite_key = base64::decode(composite_key)?;
+    if composite_key.len() < ED25519_PRIVATE_KEY_LENGTH {
+        return Err(Error::InternalError(
+            "Insufficient key length returned by vault export key request".into(),
+        ));
+    }
+    Ok(Ed25519PrivateKey::try_from(
+        &composite_key[..ED25519_PRIVATE_KEY_LENGTH],
+    )?)
```

Comment:
This looks clearer to me:

```rust
let composite_key = base64::decode(composite_key);
if let Some(composite_key) = composite_key.get(0.. ED25519_PRIVATE_KEY_LENGTH) {
  Ed25519PrivateKey::try_from(&composite_key)
} else {
 Err(...)
}
```

Reply:
or even better imo, refactor the original function to avoid the duplicate code:

```rust
if resp.not_ok() {
  return Err(resp.into());
}

let export_key: ExportKeyResponse = serde_json::from_str(&resp.into_string()?)?;
let key = if Some(version) = version {
    let key = export_key.data.keys.iter().find(|(k, _v)| **k == version);
    let (_, key) = key.ok_or_else(|| Error::NotFound("transit".into(), name.into()))?;
    key
} else if let Some(key) = export_key.data.keys.values().last() {
    key
} else {
    return Err(...);
}

let composite_key = base64::decode(composite_key)?;
if let Some(composite_key) = composite_key.get(0..ED25519_PRIVATE_KEY_LENGTH) {
  Ed25519PrivateKey::try_from(&composite_key)
} else {
  Err(...)
}
```